### PR TITLE
Staggered Release

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Merge backend PRs:
[Part 1](https://github.com/empirical-org/Empirical-Core/pull/9823): Update `previously_assigned_recommendations` and `assign_selected_packs` endpoints
[Part 2](https://github.com/empirical-org/Empirical-Core/pull/9839): Update `student_profile_data` endpoint
[Part 3](https://github.com/empirical-org/Empirical-Core/pull/9866): Add `UserPackSequenceItem`
[Part 4](https://github.com/empirical-org/Empirical-Core/pull/9936): Call `SaveUserPackSequenceItems` when activity session destroyed.
[Part 5](https://github.com/empirical-org/Empirical-Core/pull/9967): Call `SaveUserPackSequenceItems` on the rest of the relevant objects.
[Part 6](https://github.com/empirical-org/Empirical-Core/pull/9999): Call `SaveUserPackSequenceItems` on objects updated via `update_all` or `save(validate: false)`
[Part 7](https://github.com/empirical-org/Empirical-Core/pull/10036): Add batch processing of `AssignRecommendationWorkers` for independent practice activity packs.
[Part 8](https://github.com/empirical-org/Empirical-Core/pull/10039): Add redirection to activity links for students who have not unlocked a particular activity.
[Part 9](https://github.com/empirical-org/Empirical-Core/pull/10049): Change schema for `PackSequenceItem` so that it belongs to `:classroom_unit` instead of `:unit`

Also:
[do not show due date and publish date for staggered packs](https://github.com/empirical-org/Empirical-Core/pull/10053)
[fix race condition with checkbox callback](https://github.com/empirical-org/Empirical-Core/pull/10068)
[add max height to release method modal](https://github.com/empirical-org/Empirical-Core/pull/10058)

into frontend [branch](https://github.com/empirical-org/Empirical-Core/tree/feat/staggered-release-frontend)

## WHY
Staggered Release is a requested feature

## HOW
See individual PRs for detail.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Staggered-release-for-diagnostic-recommendations-7b11a96525d847f5b7e2ec5280a32f70

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

File: services/QuillLMS/app/controllers/concerns/teacher_fixes.rb [hunk 1]
```diff
@@ -105,49 +105,49 @@ def self.merge_two_schools(from_school_id, to_school_id)
     SchoolsUsers.where(school_id: from_school_id).update_all(school_id: to_school_id)
   end
 
-  def self.merge_two_classrooms(class_id1, class_id2)
```
Comment (emilia-friedberg) [inline services/QuillLMS/app/controllers/concerns/teacher_fixes.rb:108]: Not sure how these changes snuck in here 🤷‍♀️

**Review by emilia-friedberg — APPROVED** [review]
Since I wrote and/or already reviewed all of the pieces of this, I just skimmed it, but it LGTM! Yay!

File: services/QuillLMS/app/controllers/concerns/teacher_fixes.rb (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/app/controllers/concerns/teacher_fixes.rb:108]: This was just some refactoring I did while addressing [changes](https://github.com/empirical-org/Empirical-Core/pull/9999) to lock status caused by teacher fixes which use updates that don't trigger callbacks.

**Review by dandrabik — APPROVED** [review]
I've skimmed through (since I approved the individual PRs in more depth) and didn't find anything that looks off. LGTM.

Also, this is a lot of code! You should be really proud of all of the hard work that went into this feature. Not only are we releasing a highly requested feature to teachers (which will make school work on Quill less overwhelming for students), you were able to refactor a lot of potentially buggy code along the way. Thanks for all of the effort you put into this. Really well done!

**Review by anathomical — APPROVED** [review]

**Pull request merged**